mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -5,821 +5,745 @@
 //!
 //! The data layout is as follows:
 //!  - The first block of the partition is not used.
-//!  - The second block contains a MetaDataHeader (currently 72 bytes) while the
-//!    remainder of the block is padded with zeros.
-//!  - The "index" starts at the third block and contains a fixed number of
-//!    MetaDataIndexEntry entries, each of which contains the address of an
-//!    object that has been written to the partition.
-//!  - The first usable "data" block is the first block following the index
-//!    (whose size is aligned to the blocksize of the disk).
+//!  - The second block contains the MetaDataIndex (currently 80 bytes) while
+//!    the remainder of the block is padded with zeros.
+//!  - The data in the "index" starts at the third block.
+//!  - This and each subsequent block contains a single MetaDataObject entry.
```

Comment:
If I understand this correctly, writing a new entry requires update 2 different disk sectors: MetaDataIndex and the MetaDataObject sector. How do we handle scenarios when crash occurs between these 2 writes ?

Reply:
We don't handle the error explicitly. The only thing that can happen is that (assuming the index is currently full) the oldest entry will be overwritten. This would not normally be a problem if we only retrieve the most recent entry. The only issue is if we retrieve the entire contents of the index, and iven in that case we could detect this case because the generation number would be out of order. This is a minor issue for now as this functionality (retrieving all entries at once) is not used at present.